### PR TITLE
3.1 resample index procedures

### PR DESCRIPTION
**Pull request by benbc**

### CODE REVIEW

File: community/kernel/src/test/java/org/neo4j/kernel/builtinprocs/AwaitIndexProcedureTest.java [hunk 1]
```diff
@@ -240,162 +217,4 @@ public void shouldTimeoutIfTheIndexTakesTooLongToComeOnline()
...
-            throw new UnsupportedOperationException( "not implemented" );
-        }
-
-        @Override
-        public QueryRegistryOperations queryRegistration()
-        {
-            throw new UnsupportedOperationException( "not implemented" );
-        }
-    }
 }
```
Comment (systay) [inline community/kernel/src/test/java/org/neo4j/kernel/builtinprocs/AwaitIndexProcedureTest.java:220]: You should add tests for escaped labels and property names, such as `:``Label with space in it``(prop)` and `:Label(``prop key with space in it``)`

Comment (benbc) [reply community/kernel/src/test/java/org/neo4j/kernel/builtinprocs/AwaitIndexProcedureTest.java:220]: Have you seen this? https://github.com/neo4j/neo4j/pull/7929/files/0d71130426771b601e72bf956abf70ee0a1c756f#diff-4867fc5c6450f04f08b33be30d307f2b

File: community/kernel/src/test/java/org/neo4j/kernel/builtinprocs/IndexSpecifierTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    @Test
+    public void shouldParseASimpleProperty()
+    {
+        assertThat( new IndexSpecifier( ":Person(a_Name_123)" ).property(), is( "a_Name_123" ) );
+    }
+
+    @Test
+    public void shouldParseANastyLabel()
+    {
+        assertThat( new IndexSpecifier( ":`:(!\"£$%^&*( )`(name)" ).label(), is( ":(!\"£$%^&*( )" ) );
```
Comment (systay) [inline community/kernel/src/test/java/org/neo4j/kernel/builtinprocs/IndexSpecifierTest.java:51]: Ah, here they are. Never mind my previous comment :)

Comment (benbc) [reply community/kernel/src/test/java/org/neo4j/kernel/builtinprocs/IndexSpecifierTest.java:51]: Your input on whether these tests are sufficient is the main reason for my asking you to review the PR, so please let me know if you think they're not sufficient.

**Comment by zhenlineo** [comment]
+1 but need a rebase

**Pull request merged**